gprofiler/spark/spark_collector.py

```diff
@@ -0,0 +1,894 @@
+        if args:
+            for directory in args:
+                url = self._join_url_dir(url, directory)
+
+        # Add kwargs as arguments
+        if kwargs:
+            query = "&".join(["{0}={1}".format(key, value) for key, value in kwargs.items() if value is not None])
+            url = urljoin(url, "?" + query)
+
+        try:
+            self._logger.debug(f"Spark check URL: {url}")
+            response = requests.get(url, timeout=3)
+            response.raise_for_status()
+
+            return response
+
+        except (HTTPError, InvalidURL, ConnectionError, ConnectionRefusedError, Timeout, ValueError):
+            raise
+
+    def _rest_request_to_json(self, address: str, object_path: str, *args: Any, **kwargs: Any) -> Any:
+        """
+        Query the given URL and return the JSON response
+        """
+
+        response = self._rest_request(address, object_path, *args, **kwargs)
+
+        try:
+            response_json = response.json()
+        except json.decoder.JSONDecodeError:
+            self._logger.exception("JSON Parse failed.")
+            raise
+
+        return response_json
+
+    def _get_request_url(self, url: str) -> str:
+        """
+        Get the request address, build with proxy if necessary
+        """
+        parsed = urlparse(url)
+
+        _url = url
+        if not (parsed.netloc and parsed.scheme):
+            _url = urljoin(self._master_address, parsed.path)
+
+        return _url
+
+    @staticmethod
+    def _join_url_dir(url: str, *args: Any) -> str:
+        """
+        Join a URL with multiple directories
+        """
+        for path in args:
+            url = url.rstrip("/") + "/"
+            url = urljoin(url, path.lstrip("/"))
+
+        return url
+
+
+class SparkSampler(object):
+    """
+    Spark cluster metrics sampler
+    """
+
+    def __init__(
+        self,
+        sample_period: float,
+        storage_dir: str,
+        api_client: Optional[APIClient] = None,
+    ):
+        self._master_address: Optional[str] = None
+        self._spark_mode: Optional[str] = None
+        self._collection_thread: Optional[Thread] = None
+        self._logger = get_logger_adapter(__name__)
+        self._sample_period = sample_period
+        self._stop_event = Event()
+        self._spark_sampler: Optional[SparkCollector] = None
+        self._stop_collection = False
+        self._is_running = False
+        self._storage_dir = storage_dir
+        if self._storage_dir is not None:
+            assert os.path.exists(self._storage_dir) and os.path.isdir(self._storage_dir)
+        else:
+            self._logger.debug("output directory is None. Will add metrics to queue")
+        self._client = api_client
+
+    def _get_yarn_config_path(self, process: psutil.Process) -> str:
+        env = process.environ()
+        if "HADOOP_CONF_DIR" in env:
+            self._logger.debug("Found HADOOP_CONF_DIR variable.", extra={"hadoop_conf_dir": env["HADOOP_CONF_DIR"]})
```

Comment:
In gProfiler's logger we can just pass extras to the different calls, so this can be
```suggestion
            self._logger.debug("Found HADOOP_CONF_DIR variable", hadoop_conf_dir=env["HADOOP_CONF_DIR"])
```
for consistency with the rest of the code.

Reply:
Same for other `logging` calls around this module.